app/controllers/api/v1/trips_controller.rb

```diff
@@ -3,8 +3,32 @@ def create
     render json: TripFacade.new(trip_params).response, status: 201
   end
 
+  def index
+    # Lookup trip by token
+    trip = Trip.find_by(token: trip_token_params[:token])
+
+    # If no trip exists, invalid trip_token
+    unless trip
```

Comment:
This is functionally a guard clause.

After line 13, you _know_ that `trip != nil`. Thus in lines 16 and 21 that doesn't need to be checked

Reply:
I agree with you.  Initially, the subsequent checks did not contain `if trip`.  However, RSpec gives this error message if I pull it out.

```
Failures:

  1) Trips API V1 requests GET /api/v1/trips status 404 if an invalid trip token provided
     Failure/Error: if trip.pending?
     
     NoMethodError:
       undefined method `pending?' for nil:NilClass
     # ./app/controllers/api/v1/trips_controller.rb:16:in `index'
     # ./spec/requests/api/v1/get_trips_request_spec.rb:11:in `block (3 levels) in <top (required)>'

```

For some reason RSpec continues past `render json` when running the test with an invalid trip token. This was my solution or monkey-patch to avoid it but I agree it feels redundant.

Reply:
Scratch my last comment, I refactored to this:

```ruby
  def index
    # Lookup trip by token
    trip = Trip.find_by(token: trip_token_params[:token])

    # If no trip exists, invalid trip_token
    if !trip
      render json: { message: 'Invalid Token'}, status: 404

    # If valid trip AND trip.status == 'pending', POI data is not yet ready
    elsif trip.pending?
      render json: { message: 'Trip pending'}, status: 202

    # If valid trip AND trip.status == "Ready", return all POIs and Legs
    elsif trip.ready?
      render json: TripLegsFacade.new(trip).response, status: 200
    end
  end
```

Reply:
I like it, thanks for the change